src/pages/parcel/[id].tsx

```diff
@@ -0,0 +1,40 @@
+import getAllParcelData from "@/utils/parcel-fetch";
+import { GetStaticProps, GetStaticPaths, InferGetStaticPropsType } from "next";
+import Parcels from "@/components/Parcels";
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const parcelData = await getAllParcelData();
+
+  // Generate paths based on parcels
+  const paths = parcelData.map((parcel) => ({
+    params: { id: parcel.id.toString() },
+  }));
+
+  // Pre-render only these paths at build time.
+  return { paths, fallback: "blocking" }; // Use 'blocking' for ISR and better user experience
+};
```

Comment:
Could you explain what this does?

Reply:
to be honest i just followed this tutorial but I understood some of it.
https://www.youtube.com/watch?v=mAHqpdVzJmA&t=244s&ab_channel=NetNinja